projects/swimlane/ngx-charts/src/lib/common/axes/x-axis-ticks.component.ts

```diff
@@ -155,6 +198,86 @@ export class XAxisTicksComponent implements OnChanges, AfterViewInit {
     return angle;
   }
 
+  getMarkerColor(color: number): string {
+    if (
+      color == null ||
+      this.markerColors == null ||
+      this.markerColors.length === 0 ||
+      color >= this.markerColors.length
+    ) {
+      return '';
+    }
+    return `${this.markerColors[color]}`;
+  }
+
+  setMarkers(): void {
+    // hide markers out of visual range
+    this.markers.forEach(line => {
+      const p = this.adjustedScale(line.value);
+      line.active = p > 0 && p < this.width;
+    });
+
+    const markerAreas = [];
+
+    const groupedMarkers = this.markers.reduce(function(r, a) {
+      r[a.group] = r[a.group] || [];
+      r[a.group].push(a);
+      return r;
+    }, Object.create(null));
+
+    for (const group in groupedMarkers) {
+      const markerGroup = groupedMarkers[group];
+
+      // use color only if it is uniform amongst all markers in group
+      const color = markerGroup.every(item => item.color === markerGroup[0].color) ? markerGroup[0].color : null;
+
+      let min = this.adjustedScale(
+        Math.min.apply(
+          null,
+          markerGroup.map(item => item.value)
+        )
+      );
+      let max = this.adjustedScale(
+        Math.max.apply(
+          null,
+          markerGroup.map(item => item.value)
+        )
+      );
+      const markerGroupLength = markerGroup.length;
+
+      // get visual order
+      const tmin = min;
+      min = Math.min(min, max);
+      max = Math.max(tmin, max);
+
+      // clip
+      if (min < 0) {
+        min = 0;
+      }
+      if (max > this.width) {
+        max = this.width;
+      }
+      const w = max - min;
+
+      if (w > 0) {
+        const markerAreaPath = roundedRect(min, -this.gridLineHeight - 5, w, this.gridLineHeight, 0, [
+          false,
+          false,
+          false,
+          false
+        ]);
+
+        markerAreas.push({
+          markerGroupLength,
+          markerAreaPath,
+          color
+        });
```

Comment:
Try to define an Interface for these objects so we can get type info about the properties inside `markerAreas` array.

Reply:
Defined internal interface for this one, strong typing is now available